src/pyk/kcfg/kcfg.py

```diff
@@ -92,13 +120,35 @@ class KCFG(Container[Union['KCFG.Node', 'KCFG.Successor']]):
     class Node:
         id: int
         cterm: CTerm
+        attrs: frozenset[NodeAttr]
+
+        def __init__(self, id: int, cterm: CTerm, attrs: Iterable[NodeAttr] = ()) -> None:
+            object.__setattr__(self, 'id', id)
+            object.__setattr__(self, 'cterm', cterm)
+            object.__setattr__(self, 'attrs', frozenset(attrs))
 
         def to_dict(self) -> dict[str, Any]:
-            return {'id': self.id, 'cterm': self.cterm.to_dict()}
+            return {'id': self.id, 'cterm': self.cterm.to_dict(), 'attrs': self.attrs}
```

Comment:
```suggestion
            return {'id': self.id, 'cterm': self.cterm.to_dict(), 'attrs': [a.value for a in self.attrs]}
```

I guess this is why we inherit `str`, but it's probably not necessary to do that, can just use `.value()` from `Enum`.

Reply:
I was trying to have it be a list of `str`, but I went with just having the attrs in the dict be a list of `NodeAttr`, so I guess it's no longer necessary to have it inherit `str`. But if we wanted it to be a list of `str` I don't know how we get from a string back to a `NodeAttr`, while allowing for arbitrary `NodeAttr` extensions.